src/bindings/python/flux/asyncio/events.py

```diff
@@ -0,0 +1,67 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import asyncio
+
+import flux
+import flux.core.watchers
+
+from .selector import FluxSelector
+
+# Make sure submit async and we see job go through
+# Ensure the job and loop use the same handle
+handle = flux.Flux()  # pylint: disable=invalid-name
+
+
+async def submit(fluxjob):
+    """
+    Submit an async Flux Job.
+
+    Example usage
+    =============
+    # ensure you've started flux
+    import asyncio
+    from flux.asyncio import loop
+    import flux.job
+
+    fluxsleep = flux.job.JobspecV1.from_command(['sleep', '2'])
+    fluxecho = flux.job.JobspecV1.from_command(['echo', 'pancakes'])
+
+    tasks = [
+       loop.create_task(asyncio.sleep(5)),
+       loop.create_task(flux.asyncio.submit(fluxecho)),
+       loop.create_task(flux.asyncio.submit(fluxsleep)),
+    ]
+
+    asyncio.set_event_loop(loop)
+    results = loop.run_until_complete(asyncio.gather(*tasks))
+    # [JobID(456004999315456), JobID(456004999315457)]
+    """
+    uid = await flux.job.submit_async(handle, fluxjob)
+    return uid
+
+
+class FluxEventLoop(asyncio.SelectorEventLoop):
+    """
+    An asyncio loop that handles running Flux.
+    """
+
+    def __init__(self, flux_handle=None):
+        if not flux_handle:
+            flux_handle = handle
+        selector = FluxSelector(flux_handle)
+
+        # Reverse reference is needed for watchers
+        selector.loop = self
+        super().__init__(selector)
+
+
+# This loop needs to be accessible from all places!
+loop = FluxEventLoop(handle)  # pylint: disable=invalid-name
```

Comment:
I'm not sure how to go about it, because we need to have a shared loop that is importable from the flux.job code too. It would work fine if the user redefines it, but if we don't have it there, we can't do the second important. It's a bit wonky how asyncio needs to do that. Do you have an idea for how else to go about it?

Reply:
You mean because Futures reach out for the `loop` attribute? I think I have another comment about how that could be avoided.

Reply:
yep - we need to be using the same loop instance across the different files, and we need to avoid circular import. What is your idea?

Reply:
https://github.com/flux-framework/flux-core/pull/4623#discussion_r982855682

Resolving this issue (to continue it in that thread, maybe)

Reply:
Okay this is fixed to use `asyncio.get_running_loop()`